src/twin.ts

```diff
@@ -0,0 +1,25 @@
+import { Seq } from "immutable";
+
+/**
+ * Twin
+ *
+ * Given two sequences, interweave them, alternating terms
+ */
+
+// Public API
+export default (
+  a: Seq.Indexed<number>,
+  b: Seq.Indexed<number>
+): Seq.Indexed<number> => Seq<number>(generator(a, b));
+
+// Generator
+function* generator(
+  a: Seq.Indexed<number>,
+  b: Seq.Indexed<number>,
+  counter: number = 0
+): Generator<number> {
+  // I'm annoyed that this doesn't work!
+  //   yield counter % 2 === 0 ? a.get(counter / 2) : b.get((counter - 1) / 2);
+  yield counter % 2 === 0 ? a.first() : b.first();
```

Comment:
This was infuriating and vexed me for ages. Basically these Seqs-wrapping-Generators are not immutable!
If you do take(5) on one of them - the first time you call it you get the first five elements - fine. But call take(5) on the Seq a second time and you get the next five elements.

So I am using first() here as kind of a next() equivalent.

Reply:
Oh shit! Will `first` always give you the first elements? So much for immutable js!